microcosm_pubsub/dispatcher.py

```diff
@@ -81,13 +66,37 @@ def handle_message(self, media_type, message):
                 # no handlers
                 self.logger.debug("Skipping message with no registered handler: {}".format(media_type))
                 return False
-            else:
+
+            try:
                 handler_with_context = context_logger(
                     self.sqs_message_context,
                     sqs_message_handler,
                     parent=sqs_message_handler,
                 )
                 return handler_with_context(message)
+            except Exception as error:
+                # NB if possible, log with the handler's logger to make it easier
+                # to tell which handler failed in the logs.
+                logger = getattr(sqs_message_handler, "logger", None)
+                if logger is None:
+                    logger = self.logger
+
+                if isinstance(error, Nack):
+                    logger.info(
+                        "Nacking SQS message: {}".format(
+                            media_type,
+                        ),
+                        extra=self.sqs_message_context(message)
+                    )
+                else:
+                    logger.error(
```

Comment:
One other thought: before we were logging this case at info level. If we know it's an error, might as well use error level.